src/shady-css/node-factory.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import { nodeType } from './common';
+
+/**
+ * Class used for generating nodes in a CSS AST. Extend this class to implement
+ * visitors to different nodes while the tree is being generated, and / or
+ * custom node generation.
+ */
+class NodeFactory {
+  /**
+   * Creates a Stylesheet node.
+   * @param {array} rules The list of rules that appear at the top
+   * level of the stylesheet.
+   * @return {object} A Stylesheet node.
+   */
+  stylesheet(rules) {
+    return { type: nodeType.stylesheet, rules };
+  }
+
+  /**
+   * Creates an At Rule node.
+   * @param {string} name The "name" of the At Rule (e.g., `charset`)
+   * @param {string} parameters The "parameters" of the At Rule (e.g., `utf8`)
+   * @param {object=} ruleset The Ruleset node (if any) of the At Rule.
+   * @return {object} An At Rule node.
+   */
+  atRule(name, parameters, ruleset) {
+    return { type: nodeType.atRule, name, parameters, ruleset };
+  }
+
+  /**
+   * Creates a Comment node.
+   * @param {string} value The full text content of the comment, including
+   * opening and closing comment signature.
+   * @return {object} A Comment node.
+   */
+  comment(value) {
+    return { type: nodeType.comment, value };
+  }
+
+  /**
+   * Creates a Ruleset node.
+   * @param {array} rules An array of the Rule nodes found within the Ruleset.
+   * @return {object} A Ruleset node.
+   */
+  ruleset(rules) {
+    return { type: nodeType.ruleset, rules };
+  }
+
+  /**
+   * Creates a Selector node.
+   * @param {string} combinator The combinator that corresponds to the Selector
```

Comment:
is `combinator` the right name? What about simple selectors like `a`? Maybe `value` instead?